docs/getting_started/folders_and_files/derivatives.md

```diff
@@ -199,31 +201,47 @@ my_dataset/
     analysis/
   sub-01/
   ...
+  dataset_description.json
 ```
 
-A BIDS Derivatives dataset may contain references to its input datasets
-in the `sourcedata/` subdirectory:
+Disadvantage is that such organization would complicate distribution of the raw BIDS dataset
+by itself as it would require explicit exclusion of datasets within its `derivatives/` folder.
+
+A BIDS Derivative dataset may contain references to its input datasets
+(could be BIDS Raw, non-BIDS or even other BIDS Derivatives) in the `sourcedata/` subdirectory:
 
 ```bash
 my_analysis/
   sourcedata/
     raw/
+      sub-01/
+      ...
+      dataset_description.json
     preprocessed/
   sub-01/
   ...
+  dataset_description.json
 ```
 
+Disadvantage here is similar -- distribution of such BIDS Derivative dataset alone would
+require explicit exclusion of the datasets within its `sourcedata/` folder.
+
 Note that the `sourcedata/` and `derivatives/` subdirectories constitute dataset boundaries.
-Any contents of these directories may be validated independently,
-but their contents must not affect the interpretation of the nested or containing datasets.
+Any subfolders of these directories may be validated independently, if they are BIDS datasets
+which would be indicated by presence of `dataset_description.json` in them with a
+REQUIRED `"BIDSVersion"` key.
+It is important to note that their contents must not affect the interpretation of the nested
+or containing datasets.
 
-Unnested datasets are also possible. For example:
+It is also possible to completely avoid nesting of datasets by simply placing them in a folder
+containing both `sourcedata/` and `derivatives/` at the same time:
```

Comment:
For me the term nesting is a bit confusing as the example below has a set of nested directories.